src/frequenz/client/base/conversion_helper.py

```diff
@@ -0,0 +1,67 @@
+# License: MIT
+# Copyright © 2024 Frequenz Energy-as-a-Service GmbH
+
+"""Helper functions to convert to/from common python types."""
+
+from datetime import datetime, timezone
+from typing import overload
+
+# pylint: disable=no-name-in-module
+from google.protobuf.timestamp_pb2 import Timestamp
+
+# pylint: enable=no-name-in-module
+
+
+@overload
+def to_timestamp(dt: datetime) -> Timestamp:
+    """Convert a datetime to a protobuf Timestamp.
+
+    Args:
+        dt: datetime object to convert
+
+    Returns:
+        datetime converted to Timestamp
+    """
+
+
+@overload
+def to_timestamp(dt: None) -> None:
+    """Overload to handle None values.
+
+    Args:
+        dt: None
+
+    Returns:
+        None
+    """
+
+
+def to_timestamp(dt: datetime | None) -> Timestamp | None:
```

Comment:
One more, I guess we might have cases where the `datetime` can't be `None`, and using this function will introduce an annoyance of having to check for `None`, even when it shouldn't be necessary. So I wonder if it might be better to call this `to_timestamp_or_none()`. Optionally you could also add a `to_timestamp()` that doesn't take (or return) `None`, or we can add it when necessary.

Reply:
Doesn't the overload with the not-`none` parameter take care of that?

Reply:
I checked it with tests and if you pass a not-`None` variable it will also return a not-`None` variable without any extra checks required by the caller to satisfy the type system.